cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/healthcheck/AdvancedHealthCheckDialog.java

```diff
@@ -141,7 +142,12 @@ public void widgetSelected(SelectionEvent e) {
               MDC.put(InfrastructureConstants.CORRELATION_ID, UUID.randomUUID().toString());
               AdvancedHealthCheckDialog.this.healthCheck.upgradeTemplatesConfiguration(
                   AdvancedHealthCheckDialog.this.upgradeableConfigurations.get(key), BackupPolicy.BACKUP_IF_POSSIBLE);
-              refreshUI();
+              try {
+                refreshUI();
+              } catch (UpgradeTemplatesNotificationException e1) {
+                // create New UI Dialog for healthcheck
+                PlatformUIUtil.openErrorDialog("You are using an old Templates version!", e1);
+              }
```

Comment:
I have seen different catch blocks with redundant exception handling, Please make sure, that this is properly and consistently handled. I think there is even an ExceptionHandler class insider of the eclipse plugin, which you could use to handle the exception centrally and consistently sharing code.